lsp/apex_ls.lua

```diff
@@ -0,0 +1,59 @@
+--- @brief
+---
+--- https://github.com/forcedotcom/salesforcedx-vscode
+---
+--- Language server for Apex.
+---
+--- For manual installation, download the JAR file from the [VSCode
+--- extension](https://github.com/forcedotcom/salesforcedx-vscode/tree/develop/packages/salesforcedx-vscode-apex) and adjust the `apex_jar_path` appropriately.
+---
+--- ```lua
+--- vim.lsp.config('apex_ls', {
+---   apex_jar_path = '/path/to/apex-jorje-lsp.jar',
+---   apex_enable_semantic_errors = false, -- Whether to allow Apex Language Server to surface semantic errors
+---   apex_enable_completion_statistics = false, -- Whether to allow Apex Language Server to collect telemetry on code completion usage
+--- }
+---```
+---
+--- Example configuration using Mason:
+---
+---```lua
+--- vim.lsp.config('apex_ls', {
+---   apex_jar_path = vim.fn.stdpath('data') .. '/mason/share/apex-language-server/apex-jorje-lsp.jar',
+--- }
+---```
+---
+--- For a complete experience, you may need to ensure the treesitter parsers for 'apex' are installed (:TSInstall apex) as well as configure the filetype for apex (*.cls) files:
+---
+--- ```lua
+--- vim.filetype.add({
+---   pattern = {
+---     ['.*/*.cls'] = 'apex',
+---   },
+--- })
+--- ```
+
+---@type vim.lsp.Config
+return {
+  cmd = function(dispatchers, config)
+    local local_cmd = {
+      vim.env.JAVA_HOME and (vim.env.JAVA_HOME .. '/bin/java') or 'java',
+      '-cp',
+      config.apex_jar_path,
```

Comment:
Interesting approach. And since `cmd()` gained the `config` arg, this is much more ergonomic. We should keep this is mind for other configs.